lib/identity_cache.rb

```diff
@@ -131,6 +134,18 @@ def fetch_multi(*keys)
       result
     end
 
+    def allow_fetch_to_return_relation
+      self.fetch_returns_relation = true
+      yield
+      self.fetch_returns_relation = false
```

Comment:
To avoid leaky behaviour, we usually store the old value in a local variable then use an ensure block to set it back to that old value.

You could also merge this function with the following one by just parameterizing the value.  E.g. `def with_ fetch_returns_relation(value=true)`